src/neuroglancer/user_report/user_report.ts

```diff
@@ -0,0 +1,201 @@
+    };
+    const isASP = (s: string, numeric: boolean) => {
+      let reg = numeric ? /[a-zA-Z\d][a-zA-Z\d .'&]+/g : /[a-zA-Z][a-zA-Z .']+/g;
+      let match = s.match(reg);
+
+      if (match) { return match[0] === s; }
+      return false;
+    };
+    const genericBlur = (e: Event) => {
+      let self: HTMLInputElement = <HTMLInputElement>e.target;
+      let label = self.getAttribute('sName') || '';
+      self.value = self.value.trim();
+      let valid =isASP(self.value, label === 'Title');
+
+      if (valid) { this.complete[label] = true; }
+      else if (validator.isEmpty(self.value, { ignore_whitespace: true})) { this.complete[label] = false; }
+      else { self.value = self.getAttribute('oldVal') || ''; }
+
+      unDisable();
+    };
+
+    simpleInput('Name', 'form_name', { required: true, onblur: genericBlur});
+    simpleInput('Email', 'form_email', { onblur: (e: Event) => {
+      let self: HTMLInputElement = <HTMLInputElement>e.target;
+      let valid = validator.isEmpty(self.value) || validator.isEmail(self.value);
+      let normal = self.value.length ? validator.normalizeEmail(self.value) || '' : '';
+
+      if (valid) { self.value = normal; }
+      else { self.value = self.getAttribute('oldVal') || ''; }
+    }});
+
+    labelWrap('Issue Type', [
+      br(),
+      giveSimpleCheck('1', void(0), 'form_type'), ' Bug ',
+      giveSimpleCheck('2', void(0), 'form_type'), ' Suggestion'
+    ]);
+
+    simpleInput('Title', 'form_title', { required: true, onblur: genericBlur});
+
+    let
+      description = document.createElement('textarea'),
+      asterick = document.createElement('span');
+    description.id = 'form_des';
+    description.placeholder = `Well, we're waiting...`;
+    description.setAttribute('required', '');
+    description.onblur = (e: Event) => {
+      let self: HTMLInputElement = <HTMLInputElement>e.target;
+      let valid = !validator.isEmpty(self.value);
+
+      if (valid) { this.complete['description'] = true; }
+      else { this.complete['description'] = false; }
+
+      unDisable();
+    };
+    description.rows = 5;
+    description.cols = 40;
+    asterick.textContent = '*';
+    asterick.style.color =' red';
+    this.complete.description = false;
+    labelWrap('Description', [asterick, br(), description]);
+
+    labelWrap('Extra Data', [
+      br(),
+      giveSimpleCheck('', 'form_shot'), ' Submit Screenshot', br(),
+      giveSimpleCheck('', 'form_surl'), ' Submit Url Address', br()
+    ]);
+
+    let submit = document.createElement('input');
+    submit.id = 'complain';
+    submit.type = 'submit';
+    submit.disabled = true;
+    submit.onclick = (e) => {
+      if (!(<HTMLInputElement>e.target).disabled) {
+        this.submit();
+      }
+    };
+    modal.appendChild(submit);
+  }
+
+  image = '';
+  complete: LooseObject = {};
+  async submit() {
+    let
+        url = `https://script.google.com/macros/s/AKfycbzmPIJMb9z_o0_2vFdNeTIgrur_b_2tFO2A3pP9w9r7RVzub5E/exec`,
```

Comment:
Just noticed issues #300, #301 and #302. That was likely me when calling the URL manually. Make sure that the script itself is robust to garbage or even malicious input.